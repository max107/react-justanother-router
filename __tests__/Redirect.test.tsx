import * as React from "react";
import { create } from 'react-test-renderer';
import {
  createMemoryHistory,
  History,
  Location,
  Redirect,
  RouterEngine,
  RouterEngineInterface,
  RouterProvider,
  Update
} from "../src";

let locations: Location[] = [];

const history: History = createMemoryHistory();
history.listen(({ location }: Update): void => {
  locations.push(location);
});

const router: RouterEngineInterface = new RouterEngine([
  { path: '/', name: 'homepage', render: () => <div>homepage</div> },
  { path: '/foobar', name: 'foo-bar', render: () => <div>foo-bar</div> },
]);

afterEach(() => {
  locations = [];
});

test('renders without error', () => {
  expect(locations.length).toEqual(0);
  const tree = create(
    <RouterProvider history={history} router={router}>
      <Redirect to='foo-bar'/>
    </RouterProvider>
  );
  expect(tree.toJSON()).toMatchSnapshot();
  expect(locations.length).toEqual(1);
});
