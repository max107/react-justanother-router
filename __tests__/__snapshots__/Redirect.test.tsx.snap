// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error 1`] = `null`;
